json_encoding.rst

```diff
@@ -0,0 +1,62 @@
+#########################
+Raiden transport messages
+#########################
+
+Transport
+=========
+Messages are exchanged in JSON format using Matrix as a transport layer. Messages can be sent either via broadcast channel or via direct messaging
+channel.
+
+Rationale
+---------
+* plaintext JSON messages are easy to read for both humans and machines
+* JSON format is widely used and can be easily validated using JSON-schema (http://json-schema.org/)
+* JSON is easily extensible and extension can be made backward-compatible
+* Message format must not be dependent on the transport used - it should not matter whether the message is sent via Matrix or REST
+* Sender identity verification must not be part of the transport. It should be possible to submit the message over an untrusted channel or delegate.
+
+Fields validation
+-----------------
+* Receiver SHOULD use JSON schema to validate integrity of received message depending on the message type.
+* For messages that contain data later used as a smart contract parameters, receiver MUST check for overflows and conversion errors.
+* Any values to be used on-chain MUST NOT in any circuimstances be floats.
+* For ecrecover-able signatures, Sender identity SHOULD be verified.
+
+Message format
+--------------
+The message MUST contain ``body`` field. This field MUST be a JSON object, and it MAY be empty.
+Contents of the field are implementation dependent. Message MAY also contain other fields in the root JSON object, but the application MUST NOT depend on existence of these fields.
```

Comment:
I don't get why we need a body field. The body value can just be put as the root object/payload. For now, looks like this spec is just to make a `body` field which _separates_/_differentiates_ between data the application MAY care/depend, and data that it must not (metadata in root). I'd rather not force this kind of semanthic or on a specific message layout/structure. We can just make `body` the root object, new [meta]data can simply be added to the body, and the app should only care about the fields it cares. IMHO clear spec on message data itself, ignoring [meta]data which isn't in the spec, is better than adding this kind of unnecessary schema to a higher level spec, which adds complexity to only gain a specific place where one can put metadata it may be sure won't affect the app (why?) and still requires a protocol/agreement on the body format anyway.

Reply:
`body` has been removed from the message, doc needs an update. The message now looks like exactly how you proposed.